OOP/assignment3/RailwayReservationSystem.java

```diff
@@ -0,0 +1,321 @@
+			int input = scanner.nextInt();
+			Train.Type type;
+			if (input == 1)
+				type = Train.Type.PASSENGER;
+			else
+				type = Train.Type.GOODS;
+			showChart(type);
+			System.out.println("Enter from to stations");
+			String fromto = scanner.next();
+			System.out.println(fromto);
+			showChart(type, fromto);
+			System.out.println("Enter Train Name");
+			String trainName = scanner.next();
+			int noOfSeat = 0;
+			if (type.equals(Train.Type.PASSENGER)) {
+				System.out.println("Enter no of seats ");
+			} else {
+
+				System.out.println("Enter waight to be transport");
+
+			}
+			noOfSeat = scanner.nextInt();
+			if (noOfSeat <= getAvailable(trainName)) {
+				int amount = 0;
+				if (type.equals(Train.Type.PASSENGER)) {
+					amount = 100 * noOfSeat;
+				} else {
+
+					amount = 50 * noOfSeat;
+
+				}
+
+				System.out
+						.println("enter the mode of payment\n1-cc\n\2-wallet\n3-netbanking");
+				int in = scanner.nextInt();
+				if (in == 1) {
+					CcPayment ccPayment = new CcPayment();
+					ccPayment.process(amount);
+
+				} else if (in == 2) {
+					WalletPayment walletPayment = new WalletPayment();
+					walletPayment.process(amount);
+
+				} else {
+					NetBankingPayment bankingPayment = new NetBankingPayment();
+					bankingPayment.process(amount);
+
+				}
+
+				for (Train train : trainList) {
+
+					if (train.getName().equals(trainName)) {
+						train.setAvail(train.getAvail() - noOfSeat);
+						break;
+					}
+				}
+
+				printTicket(name, trainName, noOfSeat, amount);
+
+			} else {
+
+				System.out.println("Seats/load unavailable");
+
+			}
+			System.out.println("Do you want to book another ticket ?y/n");
+			String in = scanner.next();
+			if (in.equals("n")) {
+
+				break;
+			} else {
+
+				showChart(type, fromto);
+			}
+
+		}
+
+	}
+
+	public static void main(String args[]) {
+
+		prepareTrain();
+		InputProcess();
+
+	}
+
+}
+
+class Train {
+
+	private String name;
+	private String fromto;
+	private int duration;
+	private int avail;
+
+	public enum Type {
+		PASSENGER, GOODS
+	}
+
+	public Type type;
+
+	public Type getType() {
+		return type;
+	}
+
+	public void setType(Type type) {
+		this.type = type;
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public String getFromto() {
+		return fromto;
+	}
+
+	public void setFromto(String fromto) {
+		this.fromto = fromto;
+	}
+
+	public int getDuration() {
+		return duration;
+	}
+
+	public void setDuration(int duration) {
+		this.duration = duration;
+	}
+
+	public int getAvail() {
+		return avail;
+	}
+
+	public void setAvail(int avail) {
+		this.avail = avail;
+	}
+
+	public static Comparator<Train> trainDuration = new Comparator<Train>() {
+
+		public int compare(Train t1, Train t2) {
+
+			int duration1 = t1.getDuration();
+			int duration2 = t2.getDuration();
+
+			/*
+			 * For ascending order
+			 */
+			return duration1 - duration2;
+
+		}
+	};
+}
+
+class Payment {
```

Comment:
Since a 'Payment' is always of some type (at least in your program) it should be an Abstract class so that in future no body could create an instance of it.